src/misc.ts

```diff
@@ -61,6 +63,16 @@ export const last = <T>(arr: T[]) => {
 	return arr && arr[arr.length - 1];
 };
 
+export const lastPresentedSample = (samples: Sample[]): Sample | undefined => {
+	let ret: Sample | undefined = undefined;
+	samples.forEach((s) => {
+		if (!ret || s.presentationTimestamp > ret.presentationTimestamp) {
+			ret = s;
+		}
+	});
+	return ret;
+};
```

Comment:
Good that you extracted this into its own function! Few nitpicks: I'd prefer the variable be named `result` instead, and the use of a normal `for..of` loop here.